From 4c940ed93a76ffa4a2b43fd234dea42c953f8c4f Mon Sep 17 00:00:00 2001
From: Ram Dayal <ram.dayal@st.com>
Date: Tue, 19 Nov 2024 14:22:11 +0100
Subject: [PATCH 30/95] dt-bindings: usb: Add ADC based Type-C controller

This commit adds a new Device Tree binding YAML file for the
ADC monitoring with CC lines for Type-C connector.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git 065d6460a82f975b5afcacfde5045ca288b2b1a2 dt-bindings: usb: Add ADC based Type-C controller]
Change-Id: I8df18bdf348f78005540d3b0b2a38479a5f8ae34
Signed-off-by: Ram Dayal <ram.dayal@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/419572
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 .../bindings/usb/adc-usb-c-connector.yaml     | 62 +++++++++++++++++++
 1 file changed, 62 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/usb/adc-usb-c-connector.yaml

diff --git a/Documentation/devicetree/bindings/usb/adc-usb-c-connector.yaml b/Documentation/devicetree/bindings/usb/adc-usb-c-connector.yaml
new file mode 100644
index 0000000000000..100b73303c821
--- /dev/null
+++ b/Documentation/devicetree/bindings/usb/adc-usb-c-connector.yaml
@@ -0,0 +1,62 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/usb/adc-usb-c-connector.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: ADC based Type-C controller
+
+maintainers:
+  - Ram Dayal <ram.dayal@st.com>
+
+description: |
+  This binding describes the ADC based Type-C controller with ADC monitoring with CC lines.
+
+properties:
+  compatible:
+    const: adc-usb-c-connector
+
+  io-channels:
+    description: List of phandles to the IIO channels for CC1 and CC2.
+    items:
+      - description: Phandle to the IIO channel for CC1.
+      - description: Phandle to the IIO channel for CC2.
+
+  io-channel-names:
+    description: Names of the IIO channels.
+    items:
+      - const: cc1
+      - const: cc2
+
+  connector:
+    type: object
+    $ref: /schemas/connector/usb-connector.yaml#
+    description: USB Type-C connector properties.
+
+required:
+  - compatible
+  - io-channels
+  - io-channel-names
+  - connector
+
+additionalProperties: false
+
+examples:
+  - |
+    typec {
+        compatible = "adc-usb-c-connector";
+        io-channels = <&adc 0>, <&adc 1>;
+        io-channel-names = "cc1", "cc2";
+
+        connector {
+            compatible = "usb-c-connector";
+            label = "USB-C";
+            data-role = "device";
+            power-role = "sink";
+            port {
+                usb_c_port: endpoint {
+                    remote-endpoint = <&usb_c_device_port>;
+                };
+            };
+        };
+    };
-- 
2.51.0

